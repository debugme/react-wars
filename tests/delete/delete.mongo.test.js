import _ from 'lodash'
import path from 'path'
import dotenv from 'dotenv'
import mongoose from 'mongoose'

import schema from '../../schemas/CharacterSchema'
import fields from './delete.data.json'

describe('delete document from characters collection', () => {

  let Character = null

  beforeAll((done) => {
    mongoose.Promise = global.Promise
    dotenv.config({ path: path.resolve(__dirname, '..', '.env') })
    const saveCharacter = () => new Character(fields).save()
    Character = mongoose.model('Characters', schema)
    mongoose.connect(process.env.mongodbUri)
      .then(saveCharacter)
      .then(done)
  })

  afterAll((done) => {
    Character.collection.remove()
      .then(() => mongoose.connection.close())
      .then(done)
  })

  it('should allow an existing character to be deleted', (done) => {
    const { name } = fields
    const find = () => Character.findOne({ name })
    const check = (data) => expect(data).toMatchSnapshot()
    Character.findOneAndRemove({ name })
      .then(find)
      .then(check)
      .then(done)
  })

  it('should return null on delete of non-existent character', (done) => {
    const name = 'banana'
    const check = (data) => expect(data).toMatchSnapshot()
    Character.findOneAndRemove({ name })
      .then(check)
      .then(done)
  })

})