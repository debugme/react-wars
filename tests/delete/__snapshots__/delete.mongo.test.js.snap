// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`delete document from characters collection should allow an existing character to be deleted 1`] = `null`;

exports[`delete document from characters collection should return null on delete of non-existent character 1`] = `null`;
